src/components/Search.js

```diff
@@ -1,17 +1,45 @@
 import React from 'react';
+import axios from 'axios';
 
 class Search extends React.Component {
+  state = {
+    query: ''
+  }
+
+  _onChange = e => {
+    const { name, value } = e.target;
+    this.setState({
+      [name]: value
+    });
+  }
+
```

Comment:
When a user types in an input, `_onChange` gets called, linking the state to the input.  This is called 2 way binding.